homeassistant/components/homematic/cover.py

```diff
@@ -20,7 +23,10 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     devices = []
     for conf in discovery_info[ATTR_DISCOVER_DEVICES]:
-        new_device = HMCover(conf)
+        if conf[ATTR_DEVICE_TYPE] in HM_GARAGE:
```

Comment:
Why does the user need to configure this, can't we auto detect it based on available data ?

Reply:
It is auto configured; it's added automatically do the discovery data when the integration is set up. I had to pass the information from the generic homematic initialization, where all devices are detected, to the individual entity types setup. Is there a better way? As far as I could see, there is other information being added there.